src/expressions/if-expr.md

```diff
@@ -74,6 +74,12 @@ if let ("Bacon", b) = dish {
 if let ("Ham", b) = dish {
     println!("Ham is served with {}", b);
 }
+
+// Irrefutable patterns are allowed primarily to make it easier for macros to
```

Comment:
I would generally avoid describing the why in reference text; just describing the what and the how is enough imo.